### PR TITLE
[azure] add job for NAT gateway

### PR DESCRIPTION
**Pull request by jinyunma**
https://issues.redhat.com/browse/CORS-2620 CI integration for Azure NAT Gateway support

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure

**Comment by jinyunma** [comment]
Job https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/42967/rehearse-42967-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-azure/1698628092152516608 failed due to known issue https://issues.redhat.com/browse/OCPBUGS-18113

Mismatched between machines and cpms for field `zones` when IPI installation on region without AZ support, and found that control plane nodes are recreated, should not be specific issues with this job.

On region with AZ support, installation is passed.

cc @jianlinliu @gpei to review

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml [hunk 1]
```diff
@@ -1303,6 +1303,40 @@ tests:
...
+- as: azure-ipi-nat-gateway-tp-f28-destructive
+  cron: 58 3 17 * *
+  steps:
+    cluster_profile: azure-qe
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:arm64-latest
+    env:
+      BASE_DOMAIN: qe.azure.devcluster.openshift.com
+      COMPUTE_NODE_TYPE: Standard_D4ps_v5
+      E2E_RUN_TAGS: '@arm64 and @azure-ipi and @network-ovnkubernetes and not @fips'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml:1332]: Missing `and @destructive` for destructive tests.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml [hunk 2]
```diff
@@ -1303,6 +1303,40 @@ tests:
...
+      FEATURE_SET: TechPreviewNoUpgrade
+      OCP_ARCH: arm64
+      OUTBOUND_TYPE: NatGateway
+      TAG_VERSION: '@4.14'
+    test:
+    - chain: openshift-e2e-test-qe
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml:1338]: It should be `openshift-e2e-test-qe-destructive` for destructive tests.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml (hunk 1 above)
Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml]: good catch, fixed.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml (hunk 2 above)
Comment (jinyunma) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__arm64-nightly.yaml]: fixed

**Pull request merged**